tests/test_report_manager.py

```diff
@@ -55,98 +87,112 @@ def setUp(self):
                 'preferred_batch_size': [4, 8],
             }
         }
+
         mock_paths = [
             'model_analyzer.reports.report_manager',
             'model_analyzer.config.input.config_command_analyze'
         ]
         self.os_mock = MockOSMethods(mock_paths=mock_paths)
         self.os_mock.start()
-        args = [
-            'model-analyzer', 'analyze', '-f', 'path-to-config-file',
-            '--analysis-models', 'test_model'
-        ]
+        self.matplotlib_mock = MockMatplotlibMethods()
+        self.matplotlib_mock.start()
 
-        yaml_content = """
-            num_configs_per_model: 5
-            client_protocol: grpc
-            export_path: /test/export/path
-            constraints:
-              perf_latency:
-                max: 100
-        """
-        config = self._evaluate_config(args, yaml_content)
-        self.report_manager = ReportManager(config=config)
+        self.perf_config = PerfAnalyzerConfig()
 
     def test_add_results(self):
-        objective_spec = {'perf_throughput': 10}
-        self.result_comparator = ResultComparator(
-            metric_objectives=objective_spec)
+        self._setUp_managers("test_model1,test_model2")
```

Comment:
There are functions in `tests/common/test_utils.py` that construct measurements and results for you. Are you unable to use those for some reason?

Reply:
There was a problem with perf_config, but I found a way to get around it. I will use the `test_utils.py`.